### PR TITLE
Refine semantics of --parallel-workers argument

### PR DESCRIPTION
**Pull request by cottsay**
Two changes:
1. Introduce special behavior when zero is specified, which will mean that there is no limit to the number of workers and that all unblocked jobs should be executed in parallel.
2. Check for negative numbers and raise a proper usage error when specified.

### CODE REVIEW

File: colcon_parallel_executor/executor/parallel.py [hunk 1]
```diff
@@ -131,8 +140,9 @@ async def _execute(self, args, jobs, *, on_error):
             for package_name, job, _ in ready_jobs:
                 # don't schedule more jobs then workers
                 # to prevent starting further jobs when a job fails
-                if len(futures) + len(take_jobs) >= args.parallel_workers:
-                    break
+                if args.parallel_workers:
+                    if len(futures) + len(take_jobs) >= args.parallel_workers:
+                        break
```
Comment (nuclearsandwich) [inline colcon_parallel_executor/executor/parallel.py:145]: It looks intentional that this conditional _won't_ run when `args.parallel_workers` is 0 (unlimited) but I did have to stop and think it through. I don't think there's anything actionable I'm just reviewing "out loud".

**Comment by nuclearsandwich** [comment]
Are there any hard limits we need to be aware of like a max number of threads / futures in the Python runtime or will the runtime manage those for us?

**Review by nuclearsandwich — APPROVED** [review]
I have one question but it's mostly for my own education and not blocking.

This seems like a reasonable improvement to me and doesn't change any default behavior.

File: colcon_parallel_executor/executor/parallel.py (hunk 1 above)
Comment (cottsay) [reply colcon_parallel_executor/executor/parallel.py:145]: A `break` in this loop causes the executor to stop looking for new work. Skipping the conditional (and the possible break) means that the executor will pluck all available work and begin executing on it regardless of the `parallel_workers` value.

I find the logic in this loop _very_ hard to follow, so I understand the pause this change yields upon review.
